src/updater.tsx

```diff
@@ -121,15 +122,16 @@ export function outdatedListeningKeys(
   callResults: MulticallState['callResults'],
   listeningKeys: { [callKey: string]: number },
   chainId: number | undefined,
-  latestBlockNumber: number | undefined
+  latestBlockNumber: number | undefined,
+  listenerOptions?: ListenerOptions
 ): string[] {
   if (!chainId || !latestBlockNumber) return []
   const results = callResults[chainId]
   // no results at all, load everything
   if (!results) return Object.keys(listeningKeys)
 
   return Object.keys(listeningKeys).filter((callKey) => {
-    const blocksPerFetch = listeningKeys[callKey]
+    const blocksPerFetch = !!listenerOptions?.blocksPerFetch ? listenerOptions.blocksPerFetch : listeningKeys[callKey]
```

Comment:
I think at this point, the blocksPerFetch should be encoded into the state, so that you do not need to pass listenerOptions into this function.

In other words, I think you can revert L125-126, L134, L252-260, and src/updater.test.ts.

Reply:
Responded to in prev comment.